### ISSUE TITLE
Can we support AppConfig for provisioning?

### ISSUE THREAD

**Issue by n8fr8** [OpenArchive/openarchive-ios#121]
https://www.appconfig.org/ios/

The AppConfig community streamlines the adoption and deployment of mobile enterprise apps by providing developers a standard approach to app configuration and management, building upon the extensive app security and configuration frameworks available in iOS.

**Comment by n8fr8** [comment]
https://developer.apple.com/library/archive/samplecode/sc2279/Introduction/Intro.html

**Comment by tladesignz** [comment]
You want a private WebDAV server configuration be pushed via MDM ("Mobile Device Management")?

We can support that, but please note:

> Note: Sensitive information such as passwords or certificates should not be sent to the device using this approach.
> 
> Apple iOS provides built in validation of the EMM system writing to the managed configurations, however does not provide encryption of these configuration values. Apple iOS only allows a device to be associated with a single EMM server, and only this EMM server can write to the managed configurations section of the application.

Seen here: https://www.appconfig.org/ios/, at the end of the "App Configuration" section.

So, what to do with the password? Ignore the warning but allow password provisioning anyway? Provision without a password, which the user gets through other means and has to enter manually? Also no username, probably so the configuration can be used with all users. So it would be just about the server URL.

**Comment by n8fr8** [comment]
Yes, no password, but possibly username. Mainly though it would be the server URL and username. The MDM admin we talked about seemed to be using it that way already. We would also want to include values for settings in the future, such as potentially "auto-delete on upload" or other privacy/security enhancing options.

**Comment by n8fr8** [comment]
(this is coming from our potentially biggest partner who is one of the largest human rights orgs and has a fairly robust enterprise system)

**Comment by foundscapes** [comment]
@n8fr8 @tladesignz let's discuss on our forthcoming call!
